api/nodecontrol/client.go

```diff
@@ -0,0 +1,160 @@
+package nodecontrol
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	nodegen "github.com/synadia-io/nex/api/nodecontrol/gen"
+	"github.com/synadia-io/nex/node/actors"
+)
+
+var (
+	DefaultRequestTimeout = 5 * time.Second
+)
+
+type ControlAPIClient struct {
+	nc *nats.Conn
+}
+
+func NewControlApiClient(nc *nats.Conn) (*ControlAPIClient, error) {
+	return &ControlAPIClient{
+		nc: nc,
+	}, nil
+}
+
+func (c *ControlAPIClient) Auction(tags []string) (*nodegen.AuctionResponseJson, error) {
+	req := nodegen.AuctionRequestJson{}
+	req_b, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	msg, err := c.nc.Request(actors.AuctionSubject(), req_b, DefaultRequestTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := new(nodegen.AuctionResponseJson)
+	err = json.Unmarshal(msg.Data, resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (c *ControlAPIClient) Ping() (*nodegen.NodePingResponseJson, error) {
+	msg, err := c.nc.Request(actors.PingSubject(), nil, DefaultRequestTimeout)
+	if err != nil {
+		return nil, err
+	}
+	resp := new(nodegen.NodePingResponseJson)
+	err = json.Unmarshal(msg.Data, resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (c *ControlAPIClient) FindAgent() (*nodegen.AgentPingResponseJson, error) {
+	msg, err := c.nc.Request(actors.AgentPingSubject(), nil, DefaultRequestTimeout)
```

Comment:
we can't use `AgentPingSubject()` here because that is defined as `AgentPingSubject NexSubject = func(_ ...string) string { return APIPrefix + ".APING.>"}`

You can't make a request on `.APING.>`

When making a request, this can be `APING.{namespace}` (which retrieves all agents in the namespace), or it can be `APING.{namespace}.{workload}` where `workload` can be the workload name (potentially returning multiple results) or the workload Id, which will return 1 or 0 results.